src/archivist/recorder/record.js

```diff
@@ -25,10 +32,14 @@ export default class Record {
     this.#content = content;
   }
 
-  static validate(givenParams) {
-    for (const param of Record.REQUIRED_PARAMS) {
-      if (!Object.prototype.hasOwnProperty.call(givenParams, param) || givenParams[param] == null) {
-        throw new Error(`"${param}" is required`);
+  validate() {
+    for (const requiredParam of this.constructor.REQUIRED_PARAMS) {
+      if (requiredParam == 'content') {
```

Comment:
Extract this check from the loop and explicitly check for `this.content`; or couldn't this be checked just as efficiently by adding `|| this[requiredParam] === ''` line 41? 🙂

Reply:
None of the solutions you propose work, because `content` is not a "data property" but an "accessor property". As it's defined with a setter and accessed with a getter, the property is not stored in the object's internal data property, but rather in the accessor property descriptor. So, `Object.prototype.hasOwnProperty` will return false.